docs/reference/mapping/runtime.asciidoc

```diff
@@ -767,6 +764,167 @@ address.
 // TESTRESPONSE[s/"_id" : "oWs5KXYB-XyJbifr9mrz"/"_id": $body.hits.hits.0._id/]
 // TESTRESPONSE[s/"day_of_week" : \[\n\s+"Sunday"\n\s\]/"day_of_week": $body.hits.hits.0.fields.day_of_week/]
 
+[[runtime-search-flattened]]
+==== Retrieve data from flattened fields
+You can retrieve values from sub-fields of
+<<search-fields-flattened,flattened fields>> with a runtime field. Instead of
+including `doc['<field_name>'].value` in your Painless script, use
+`doc['<field_name>.<sub-field_name>'].value`. For example, if you have a
```

Comment:
I am not super sure that this whole docs belong to the runtime fields docs. It certainly applies to runtime fields and helps users, but accessing fields from a script is something more general and applies broadly to scripts. I don't know if my expectations are right, but I would expect the flattened field docs to clarify how sub-fields are accessed, and accessing them from a script is the same. Otherwise we end up needing to document how to access flattened subfields, or any other special field, in every section of the search request?

Reply:
I also think this clarification fits best in the `flattened` field docs. We made a similar choice when we added support for fetching flattened fields in #70916. It's great to keep details about specific field type behavior in that field type's docs when possible, to keep cross-cutting documentation like this more streamlined.

Reply:
I think that I agree. Perhaps it belongs in the docs for flattened fields? And as we find situations like this, we can enhance the docs for a particular field type to indicate how to retrieve data using runtime fields. Does that approach make sense @javanna?